packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3287,6 +3287,22 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @mustCallSuper
   void didChangeDependencies() {
     assert(_active); // otherwise markNeedsBuild is a no-op
+    assert(() {
+      if (owner._debugCurrentBuildTarget == null) {
+        throw new FlutterError(
+          'didChangeDependencies for ${widget.runtimeType} was called at an '
+          'inappropriate time. It may only be called while the widgets are '
```

Comment:
I think the preferred format for FlutterError messages is to insert a \n after the first summary statement.